lambeq/rewrite/base.py

```diff
@@ -393,3 +399,16 @@ def _ar(self, box: Box) -> Diagram:
 
     def _ob(self, ob: Ty) -> Ty:
         return ob
+
+
+class ImperativeRewriteRule():
```

Comment:
It would be better to design a new abstract class for diagram level rewriting and inherit from it here.